classes/Carrier.php

```diff
@@ -959,8 +979,9 @@ public function addDeliveryPrice($price_list, $delete = false)
                 Db::getInstance()->execute(
                     'DELETE FROM `' . _DB_PREFIX_ . 'delivery`
                     WHERE ' . (null === $values['id_shop'] ? 'ISNULL(`id_shop`) ' : 'id_shop = ' . (int) $values['id_shop']) . '
-                    AND ' . (null === $values['id_shop_group'] ? 'ISNULL(`id_shop`) ' : 'id_shop_group=' . (int) $values['id_shop_group']) . '
-                    AND id_carrier=' . (int) $values['id_carrier'] .
+                    AND ' . (null === $values['id_shop_group'] ? 'ISNULL(`id_shop_group`) ' : 'id_shop_group=' . (int) $values['id_shop_group']) .
+                    // delete delivery records related to old carrier records
+                    'AND id_carrier IN (' . implode('","', $old_carrier_record_ids) .
```

Comment:
` 'AND id_carrier IN (` 
I can't find the closing parenthesis (maybe I missed it?)